app/controllers/volunteers_controller.rb

```diff
@@ -10,53 +12,58 @@ def index
     # profile page.
     # @repair = Repair.all
     @volunteer = Volunteer.all
+    authorize Volunteer
   end
 
   def new
-    # @repair = Repair.all
-    @volunteer = Volunteer.new
-    unless current_user.volunteer.nil?
-      redirect_to volunteer_path(current_user.volunteer.id)
-    end
-  end
-
-  def show
-    @volunteer = current_user.volunteer
-  end
-
+    @volunteer = current_user.build_volunteer
+    authorize @volunteer
+  end 
+  
   def create
-    # @repair = Repair.all
-    @volunteer = Volunteer.new(volunteer_params)
-    @volunteer.user_id = current_user.id
+    @volunteer = current_user.build_volunteer(volunteer_params)
+    authorize @volunteer
     if @volunteer.save
-      flash[:success] = 'Volunteer created!'
+      flash[:success] = "Volunteer created."
       redirect_to volunteer_path(@volunteer)
     else
       render 'new'
     end
   end
 
+  def show
+    @volunteer = current_user.volunteer unless current_user.admin?
+    @volunteer = Volunteer.find(params[:id])
```

Comment:
These definitions for `@volunteer` seem contradictory. No matter what happens in line 35, line 36 will redefine `@volunteer` based on the current `id` parameter.

Reply:
Yeah, I doubted that. Thanks Leila! I'll look deeper into it tomorrow.

Reply:
Oh, that's right.